verification/curator-service/api/src/controllers/auth.ts

```diff
@@ -158,59 +207,117 @@ export class AuthController {
             },
         );
 
+        /**
+         * Generate reset password link
+         */
         this.router.post(
-            '/signup',
+            '/request-password-reset',
             async (req: Request, res: Response): Promise<void> => {
-                let user = await User.findOne({
-                    email: req.body.email,
-                }).exec();
+                const email = req.body.email as string;
 
-                // Update newsletter preferences
-                if (
-                    user &&
-                    user.newsletterAccepted === false &&
-                    req.body.newsletter
-                ) {
-                    user = await User.findOneAndUpdate(
-                        { email: req.body.email },
-                        { newsletterAccepted: req.body.newsletter },
+                try {
+                    // Check if user with this email address exists
+                    const user = await User.findOne({ email });
+                    if (!user) {
+                        throw new Error('User does not exist');
```

Comment:
It's best not to make an error for this case, because it can be used to quickly see if another person has an account with us or to build a database of users at common domains like Gmail or live.com. More info here: https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html

By all means show an error if something goes wrong, but return 200 whether or not the user exists. The UI message should be something non-committal like "if you have an account with us, check your email for reset instructions."

Reply:
Got it, thanks for the clarification, I will change this tomorrow